### PR TITLE
add strict and force args to skip prompts and stop on failure.

### PR DESCRIPTION
**Pull request by Yaakov-weiss**
This is for automating use of the script.
it allows us to ski the prompts and stop in case of a failure.

### CODE REVIEW

File: importing/import_objects.py [hunk 1]
```diff
@@ -309,17 +309,24 @@ def add_object(line, counter, position_decrement_due_to_rule, position_decrement
...
         elif "will place the exception in an Exception-Group" in api_reply.error_message:
             return add_object(line, counter, position_decrement_due_to_rule - 1, position_decrement_due_to_section,
                               fields, api_type, generic_type, layer, layers_to_attach,
-                              changed_layer_names, api_call, num_objects, client)
+                              changed_layer_names, api_call, num_objects, client, args)
 
         position_decrement_due_to_rule += 1
 
         debug_log(log_err_msg, True, True)
+        if args.strict:
```
Comment (chkp-royl) [inline importing/import_objects.py:323]: what if args is None? better to add check for this

File: utils.py [hunk 2]
```diff
@@ -64,6 +64,10 @@ def populate_parser(parser):
                         help="UNSAFE! Auto accept fingerprint during certificate verification.")
     parser.add_argument("-t", "--session-timeout", required=False,
                         help="Session expiration timeout in seconds.")
+    parser.add_argument("--force", required=False, default=False, action="store_true",
```
Comment (chkp-royl) [inline utils.py:67]: There is also fingerprint confirmation message as part of Python SDK, Look for unsafe_auto_accept in APIClientArgs
Add warning in flag description so user be aware it's override --unsafe-auto-accept flag.

**Review by chkp-royl — COMMENTED** [review]
Hi,
I left few comments.
Please fix them and I will review it again.

Thanks,
Roy

File: importing/import_objects.py (hunk 1 above)
Comment (Yaakov-weiss) [reply importing/import_objects.py]: Done

File: utils.py (hunk 2 above)
Comment (Yaakov-weiss) [reply utils.py:67]: Added

File: importing/import_objects.py [hunk 3]
```diff
@@ -320,7 +320,7 @@ def add_object(line, counter, position_decrement_due_to_rule, position_decrement
         position_decrement_due_to_rule += 1
 
         debug_log(log_err_msg, True, True)
-        if args.strict:
+        if args is not None and args.strict:
```
Comment (chkp-royl) [inline importing/import_objects.py:323]: V

File: utils.py [hunk 4]
```diff
@@ -65,7 +65,7 @@ def populate_parser(parser):
     parser.add_argument("-t", "--session-timeout", required=False,
                         help="Session expiration timeout in seconds.")
     parser.add_argument("--force", required=False, default=False, action="store_true",
-                        help="Force run the command with no confirmation.")
+                        help="Force run the command with no confirmation. WARNING! - this will set unsafe-auto-accept to be true as well.")
```
Comment (chkp-royl) [inline utils.py:68]: You need to add logic to override unsafe-auto-accept flag before initialize APIClientArgs

**Review by chkp-royl — COMMENTED** [review]
See my comments.
Roy

File: import_export_package.py [hunk 5]
```diff
@@ -24,6 +24,8 @@ def get_version(client):
 
     arg_parser = argparse.ArgumentParser(description="R80.X Policy Package Export/Import Tool, V3.0")
     args = process_arguments(arg_parser)
+    if args.force
+        args.unsafe_auto_accept = "true"
```
Comment (chkp-royl) [inline import_export_package.py:28]: args.unsafe_auto_accept is boolean. Please change to True instead of "true"

File: import_export_package.py [hunk 6]
```diff
@@ -25,7 +25,7 @@ def get_version(client):
     arg_parser = argparse.ArgumentParser(description="R80.X Policy Package Export/Import Tool, V3.0")
     args = process_arguments(arg_parser)
     if args.force
-        args.unsafe_auto_accept = "true"
+        args.unsafe_auto_accept = True
```
Comment (chkp-royl) [inline import_export_package.py:28]: V

**Pull request merged**